src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -140,7 +140,27 @@ def _bigquery(self) -> bigquery.Client:
             return bigquery.Client(project=self._source.project)
 
     def _run_sql(self, query: str) -> BigQueryRows:
-        return self._bigquery.query(query)
+        attempt = 0
+        backoff_intervals = (10, 20, 40, 80)
```

Comment:
You have the option of storing the length in a variable, which could be re-used throughout the method on place of the calls to `len`. This would also facilitate expressing the backoffs formulaically as `tuple(10*2**i for i in range(num_retries))`. But I'll leave it up to you, I might be over-engineering here.

Reply:
i was suggested to switch over from a formula to a clearly defined set of tuples, but I'll take up your suggestion of storing the len in a var.

Reply:
I was planning on using the suggestion of a variable for the length, but after incorporating the suggestions below, it would only be used twice. I'm going to keep it out.

Reply:
Yeah, var isn't worth it if you're sticking with the tuple literal. All good!